### PR TITLE
airbrake-ruby: fix add_filter block API

### LINKED ISSUE #8: add_filter throwing exception
Migrating from 4.3.4 to 5.0.1 and when I try to configure Airbrake using an inline add_filter, I'm getting an exception.

I've got the add_filter just like in the migration_guide for testing purposes:
```
Airbrake.add_filter do |notice|
  if notice[:errors].any? { |error| error[:type] == 'RuntimeError' }
    notice.ignore!
  end
end
```

In a rails console I try this:
```
2.1.6 :003 > e = Exception.new("Hello World")
 => #<Exception: Hello World> 
2.1.6 :004 > Airbrake.notify(e)
NoMethodError: undefined method `call' for nil:NilClass
	from /Users/jschuman/.rvm/gems/ruby-2.1.6@kona/gems/airbrake-ruby-1.0.0/lib/airbrake-ruby/filter_chain.rb:64:in `block in refine'
	from /Users/jschuman/.rvm/gems/ruby-2.1.6@kona/gems/airbrake-ruby-1.0.0/lib/airbrake-ruby/filter_chain.rb:62:in `each'
	from /Users/jschuman/.rvm/gems/ruby-2.1.6@kona/gems/airbrake-ruby-1.0.0/lib/airbrake-ruby/filter_chain.rb:62:in `refine'
	from /Users/jschuman/.rvm/gems/ruby-2.1.6@kona/gems/airbrake-ruby-1.0.0/lib/airbrake-ruby/notifier.rb:123:in `send_notice'
	from /Users/jschuman/.rvm/gems/ruby-2.1.6@kona/gems/airbrake-ruby-1.0.0/lib/airbrake-ruby/notifier.rb:45:in `notify'
	from /Users/jschuman/.rvm/gems/ruby-2.1.6@kona/gems/airbrake-ruby-1.0.0/lib/airbrake-ruby.rb:280:in `call_notifier'
	from /Users/jschuman/.rvm/gems/ruby-2.1.6@kona/gems/airbrake-ruby-1.0.0/lib/airbrake-ruby.rb:134:in `notify'
	from (irb):4
	from /Users/jschuman/.rvm/gems/ruby-2.1.6@kona/gems/railties-4.2.5/lib/rails/commands/console.rb:110:in `start'
	from /Users/jschuman/.rvm/gems/ruby-2.1.6@kona/gems/railties-4.2.5/lib/rails/commands/console.rb:9:in `start'
	from /Users/jschuman/.rvm/gems/ruby-2.1.6@kona/gems/railties-4.2.5/lib/rails/commands/commands_tasks.rb:68:in `console'
	from /Users/jschuman/.rvm/gems/ruby-2.1.6@kona/gems/railties-4.2.5/lib/rails/commands/commands_tasks.rb:39:in `run_command!'
	from /Users/jschuman/.rvm/gems/ruby-2.1.6@kona/gems/railties-4.2.5/lib/rails/commands.rb:17:in `<top (required)>'
	from bin/rails:4:in `require'
	from bin/rails:4:in `<main>'
```

If I switch the filter to be a *class* and new up that class in the add_filter call then it works fine.

### PR DESCRIPTION
**Pull request by kyrylo**
Fixes #8 (add_filter throwing exception)

### CODE REVIEW

File: spec/airbrake_spec.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@
   end
 
   before do
-    described_class.configure do |c|
+    @notifier = described_class.configure do |c|
```
Comment (zefer) [inline spec/airbrake_spec.rb]: Would this be more consistent as a `let` or `subject`?

Comment (kyrylo) [inline spec/airbrake_spec.rb]: `let` or `subject` cache their values, don't they? Here we need to reset the notifier to make sure that we always use a fresh one.

**Comment by kyrylo** [comment]
PTAL

File: spec/airbrake_spec.rb (hunk 1 above)
Comment (zefer) [inline spec/airbrake_spec.rb]: Cached only within that test https://www.relishapp.com/rspec/rspec-core/v/2-11/docs/helper-methods/let-and-let

Comment (kyrylo) [inline spec/airbrake_spec.rb]: Awesome, thanks. Replaced with `let!`. PTAL

File: spec/airbrake_spec.rb [hunk 2]
```diff
@@ -5,12 +5,14 @@
     'https://airbrake.io/api/v3/projects/113743/notices?key=fd04e13d806a90f96614ad8e529b2822'
   end
 
-  before do
+  let!(:notifier) do
```
Comment (zefer) [inline spec/airbrake_spec.rb]: I don't think you need the `!`, here, unless I'm missing something?

Comment (kyrylo) [inline spec/airbrake_spec.rb]: Otherwise some tests start failing.

Comment (zefer) [inline spec/airbrake_spec.rb]: I see it now :-)

**Comment by zefer** [comment]
LGTM

**Pull request merged**